examples/bzlmod/cross_compile/README.md

```diff
@@ -38,113 +41,95 @@ The setup requires three steps, first declare dependencies and toolchains in you
+    name = "sysroot_linux_aarch64",
+    build_file = "//build/sysroot:BUILD.bazel",
+    sha256 = "d303cf3faf7804c9dd24c9b6b167d0345d41d7fe4bfb7d34add3ab342f6a236c",
+    urls = ["https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/906cc7c6bf47d4bd969a3221fc0602c6b3153caa/debian_bullseye_arm64_sysroot.tar.xz"],
 )
 ```
 
-Here, we declare to new http downloads that retrieve the sysroot for linux_x64 (Intel/AMD) and linux_aarch64 (ARM/Apple Silicon). Note, these are only
-sysroots, that means you have to configure LLVM next to use these files. As mentioned earlier, three LLVM toolchains
-needs to be configured and to do that, please add the following to your MODULE.bazel
+Here, we declare to new http downloads that retrieve the sysroot for linux_x64 (Intel/AMD) and linux_aarch64 (ARM/Apple Silicon). The buildfile is a simple filegroup and located in the /build/sysroot directory. You have to copy it in your project directory to make the sysroots work. Note, these are only
+sysroots, that means you have to configure LLVM next to use these files. 
 
-```Starlark
-LLVM_VERSIONS = {
-    "": "16.0.0",
-}
+If you need a custom sysroot, for example to cross compile system dependencies such as openssl, 
+libpq (postgres client library) or similar, read through the excellent tutorial by Steven Casagrande:
 
-# Host LLVM toolchain.
-llvm.toolchain(
-    name = "llvm_toolchain",
-    llvm_versions = LLVM_VERSIONS,
-)
-use_repo(llvm, "llvm_toolchain", "llvm_toolchain_llvm")
+https://steven.casagrande.io/posts/2024/sysroot-generation-toolchains-llvm/
+
+As mentioned earlier, three LLVM targets need to be configured and to do just that, 
+please add the following to your MODULE.bazel
 
-# X86 LLVM Toolchain with sysroot.
-# https://github.com/bazel-contrib/toolchains_llvm/blob/master/tests/WORKSPACE.bzlmod
+```Starlark
+llvm = use_extension("@toolchains_llvm//toolchain/extensions:llvm.bzl", "llvm", dev_dependency = True)
 llvm.toolchain(
-    name = "llvm_toolchain_x86_with_sysroot",
-    llvm_versions = LLVM_VERSIONS,
+    name = "llvm_toolchain",
+    llvm_version = "16.0.0",  # Same LLVM version for all platforms
+    stdlib = {
+        "linux-x86_64": "stdc++",
+        "linux-aarch64": "stdc++",
+    },
 )
 llvm.sysroot(
-    name = "llvm_toolchain_x86_with_sysroot",
-    label = "@org_chromium_sysroot_linux_x64//:sysroot",
+    name = "llvm_toolchain",
+    label = "@sysroot_linux_x64//:sysroot",
     targets = ["linux-x86_64"],
 )
-use_repo(llvm, "llvm_toolchain_x86_with_sysroot")
-
-#
-# ARM (aarch64) LLVM Toolchain with sysroot.
-# https://github.com/bazelbuild/rules_rust/blob/main/examples/bzlmod/cross_compile/WORKSPACE.bzlmod
-llvm.toolchain(
-    name = "llvm_toolchain_aarch64_with_sysroot",
-    llvm_versions = LLVM_VERSIONS,
-)
 llvm.sysroot(
-    name = "llvm_toolchain_aarch64_with_sysroot",
-    label = "@org_chromium_sysroot_linux_aarch64//:sysroot",
+    name = "llvm_toolchain",
```

Comment:
I don't quite understand how the new version works: naively this has the same name as the previosly-defined llvm_toolchain, so it will create a name collision?

Reply:
No it doesn't. I got confused by this as well since the WORKSPACE config is quite different. Previously, I declared 3 different LLVM, one for host and 2 for the target platform, but that caused to download LLVM 3 times... 
I asked another engineer who is more senior and who pointed out, I actually can just declare one, add the sysroot to it, and then it works as good and that is exactly what I am doing here. I don't exactly understand the details either, but in my experience, it works just fine in 3 different projects.